crates/apollo-compiler/src/validation/mod.rs

```diff
@@ -420,12 +421,28 @@ impl DiagnosticList {
         })
     }
 
+    /// Concatenate an `other` list of diagnostics into `self`, and sort them together.
+    pub fn merge(&mut self, other: Self) {
+        if !Arc::ptr_eq(&self.sources, &other.sources) {
+            let sources = Arc::make_mut(&mut self.sources);
+            for (&k, v) in &*other.sources {
+                sources.entry(k).or_insert_with(|| v.clone());
+            }
+        }
```

Comment:
Yes this sounds useful but it can’t be an inherent method of `SourceMap` which is an alias for `Arc<IndexMap<…>>`. Where would be a good place for it?

Reply:
Let’s consider that in https://github.com/apollographql/apollo-rs/pull/747